src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml

```diff
@@ -4273,6 +4379,64 @@ components:
             validators:
               $ref: "#/components/schemas/ValidatorUptimeCollection"
 
+    #
+    # /wallet/* Request & Response Types
+    #
+
+    WalletTbdRequest:
+      allOf:
+        - $ref: "#/components/schemas/AtLedgerStateMixin" # TODO do we even want to support that? if yes, it could technically fall under regular /state/* endpoints
+        - type: object
+          required:
+            - lookup
+          properties:
+            lookup:
+              type: array
+              items:
+                $ref: "#/components/schemas/WalletTbdRequestLookupTuple"
+      example:
+        lookup:
+          - account_locker_address: TBD
+          - account_address: TBD
+    WalletTbdRequestLookupTuple:
+      allOf:
+        - $ref: "#/components/schemas/FromLedgerStateMixin"
+        - type: object
+          required:
+            - account_locker_address
+            - account_address
+          properties:
+            account_locker_address:
+              $ref: "#/components/schemas/Address"
+            account_address:
+              $ref: "#/components/schemas/Address"
+
+    WalletTbdResponse:
+      allOf:
+        - $ref: "#/components/schemas/LedgerStateMixin"
+        - type: object
+          required:
+            - items
+          properties:
+            items:
+              type: array
+              items:
+                $ref: "#/components/schemas/WalletTbdResponseItem"
+    WalletTbdResponseItem:
+      type: object
+      required:
+        - account_locker_address
+        - account_address
+        - last_updated_at_state_version
```

Comment:
I don't think we should call it `last_updated_at_state_version`. I think it should perhaps be called `resource_last_stored_at_state_version` - and also it needs to be optional - if an `(account_locker_address, account_address)` tuple has never had any resources stored (which will be the most common case).

Or do you intend to not return such a state?

I think I'd personally prefer returning an option for each requested address, so it can be zipped with the request - but if you feel strongly for us not to, that's okay. Either way, we should document the endpoint in the description.

Reply:
1. Renamed.
2. The idea is not to return "missing" tuples. This is the model we use elsewhere. What's more it reduces number of null checks / unnecessarily nullable properties.

Reply:
> This is the model we use elsewhere. What's more it reduces number of null checks / unnecessarily nullable properties.

Can I just ask for "where" it's used elsewhere when we have an explicit batch lookup? Is that our approach in e.g. metadata?

For endpoints in the new sub-api, I'd rather we go back to the drawing board and use the pattern we think is best (easiest to use for a client and most flexible for extension or changes in the backend). I think the zippable approach wins on both fronts.

If it makes it conceptually nicer, I think we should even consider this kind of format `{ key: ObjectX, value: ObjectY? }` for these kinds of explicit lookup requests against some index.

In this case, we'd get structures like this: `{ key: { account_locker_address, account_address }, value: None }` and `{ key: { account_locker_address, account_address }, value: { last_updated_at_state_version } }`